server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/AbstractRecipientRewriteTableMailet.java

```diff
@@ -123,75 +127,158 @@ public void service(Mail mail) throws MessagingException {
      * @param message
      *            the mail message
      * @return a collection of mapped recpient addresses
-     * 
+     *
      * @throws MessagingException
      */
-    protected Collection<MailAddress> handleMappings(Mappings mappings, MailAddress sender, MailAddress recipient, MimeMessage message) throws MessagingException {
-        Iterator<Mapping> i = mappings.iterator();
-        Collection<MailAddress> remoteRecipients = new ArrayList<MailAddress>();
-        Collection<MailAddress> localRecipients = new ArrayList<MailAddress>();
-        while (i.hasNext()) {
-            Mapping rcpt = i.next();
-
-            if (!rcpt.hasDomain()) {
-                // the mapping contains no domain name, use the default domain
+    protected Collection<MailAddress> handleMappings(final Mappings mappings, MailAddress sender, MailAddress recipient, MimeMessage message) throws MessagingException {
+        ImmutableList<Mapping> nonDomains = FluentIterable.from(mappings)

```

Comment:
"Mapping" object can be address, domain ..., in this case it is address. But some addresses can miss domain part then it should use the default domain. So it needs to filter those "mapping" object and then append its domain.
I changed the variable name to make it sence